RemoteSpiNNaker/RemoteSpiNNakerJobProcessManager/src/main/java/uk/ac/manchester/cs/spinnaker/job_parameters/GitPyNNJobParametersFactory.java

```diff
@@ -45,11 +48,18 @@ public JobParameters getJobParameters(final Job job,
         }
     }
 
-    /** Constructs the parameters by checking out the git repository. */
+    /** Constructs the parameters by checking out the git repository */
     private JobParameters constructParameters(final Job job,
             final File workingDirectory, final String experimentDescription)
-            throws GitAPIException, InvalidRemoteException, TransportException {
+            throws GitAPIException, InvalidRemoteException, TransportException,
+            URISyntaxException {
         final CloneCommand clone = cloneRepository();
+        URIish urish = new URIish(experimentDescription);
+        if (urish.getUser() != null) {
+            clone.setCredentialsProvider(
+                new UsernamePasswordCredentialsProvider(
+                    urish.getUser(), urish.getPass()));
```

Comment:
What happens if the user gives a username but no password? Is the failure message at least sensible?

Reply:
The answer to this was "you get a NullPointerException" but I have now added a check which replaces the password with an empty string.  Unless the repository is set up to not care about the password this will likely fail, but at least it fails with "not authorized" which is more acceptable.